src/encoded/static/components/navigation/components/LeftNav.js

```diff
@@ -166,3 +171,181 @@ const DataNavItemBody = React.memo(function DataNavItemBody(props) {
+            if (evtKey === 'ByAccession') {
+                isValid = object.isAccessionRegex(searchText);
+            }
+            if (searchInputIsValid !== isValid) {
+                setSearchInputIsValid(isValid);
+            }
+        }
+    });
+    //render hidden form inputs
+    const renderHiddenInputsForURIQuery = useCallback(function () {
+        const query = {};
+        switch (searchItemType) {
+            case 'ExperimentSetReplicate': {
+                const browseBaseParams = navigate.getBrowseBaseParams();
+                _.extend(query, browseBaseParams);
+            }
+                break;
+            case 'ByAccession':
+                break;
+            default: {
+                _.extend(query, { 'type': searchItemType });
+            }
+                break;
+        }
+        return SearchBar.renderHiddenInputsForURIQuery(query);
+    });
+    //custom placeholder text for search item type
+    const getSearchTextPlaceholder = useCallback(function () {
+        switch (searchItemType) {
+            case 'Item':
+                return 'Search 4DN Data Portal';
+            case 'ByAccession':
+                return 'Type Item\'s Accession (e.g. 4DNXXXX ...)';
+            default:
+                return "Search in " + AvailableSearchItemTypes[searchItemType].text;
+        }
+    });
+    //navigate to Item page directly without searching
+    const navigateByAccession = useCallback(function (evt) {
+        if (searchItemType === 'ByAccession') {
+            evt.preventDefault();
+            evt.stopPropagation();
+            navigate('/' + searchText);
+        }
+    });
+    const handleOnChange = useCallback(function (evt) {
+        const value = evt.target.value;
+        setSearchText(value);
+
+        //validate accession
+        let isValid = true;
+        if (searchItemType === 'ByAccession') {
+            isValid = object.isAccessionRegex(value);
+        }
+        if (searchInputIsValid !== isValid) {
+            setSearchInputIsValid(isValid);
+        }
+    });
+    //select all text when focused
+    const handleFocus = useCallback(function (evt) {
+        evt.target.select();
+    });
+
+    const selectedItem = AvailableSearchItemTypes[searchItemType];
+    const action = (selectedItem && selectedItem.action) || '/search';
+    const btnIconClassName = 'icon icon-fw fas ' + (searchItemType === 'ByAccession' ? 'icon-arrow-right' : 'icon-search');
+    const btnDisabled = !(searchText &&  typeof searchText === 'string' && searchText.length > 0);
+    const searchTextClassName = 'form-control search-query w-100' + (!searchInputIsValid ? ' border border-danger' : '');
+
+    return (//Form submission gets serialized and AJAXed via onSubmit handlers in App.js
+        <React.Fragment>
+            <h4>Search</h4>
+            <form action={action} method="GET" className="form-inline navbar-search-form-container" onSubmit={navigateByAccession}>
```

Comment:
I.e. can get rid of form-inline here. Form-inline is for if you are going to lay out the children as display: inline items, but you're using flex (via className="row"), so inline is unnecessary.